### PR TITLE
feat: port ai-pr-review v0.7.0 improvements

### PR DESCRIPTION
**Pull request by gchaix**
## Summary

Ports applicable improvements from ai-pr-review v0.7.0 into the comprehensive-review Claude Code plugin. Changes that don't apply to the Claude Code execution model (prompt caching API, container images, fork PR support, cache priming) are excluded.

### Phase 1 — Knowledge-cutoff guardrail standardization
- **Bug fix:** adversarial-general was missing the version-hallucination guardrail entirely
- Upgraded all 5 custom findings agents to use a standardized guardrail block adapted from ai-pr-review's `_knowledge-cutoff.md` trailer
- New text includes: 4-item whitelist of valid circumstances to flag, concrete "DO NOT emit" examples, and the critical "omit the finding entirely" instruction
- edge-case-hunter gets a 5th item specific to its path-tracing mission

### Phase 2 — Trufflehog test-file demotion
- Unverified secrets in test/fixture/example files demoted from High/85 to Low/40 with `[test file — likely mock data]` tag
- Verified secrets in test files remain Critical/95 (never demoted)
- Replaced per-file trufflehog loop with single invocation (N-1 fewer process startups)
- Added `trufflehog-unverified-test-file` suppression rule as safety net

### Phase 3 — Ruby-org verify type
- Added `ruby-org` as a supported verify ecosystem for version-hallucination suppressions
- Checks ruby-lang.org's canonical tarball index via HEAD request

### Phase 4 — Checkov IaC scanner integration
- New `run-checkov.sh` covering Terraform, k8s YAML, Dockerfiles, CloudFormation, Azure ARM
- Content-sniff for YAML/JSON avoids checkov's ~3-8s Python cold start on non-IaC files
- Severity mapping: `CKV2_*`/`CKV_SECRET_*` → High, others → Medium
- Opportunistic — silently skips if checkov binary not installed

### Not ported (with rationale)
| Feature | Why |
|---|---|
| Prompt caching API (shared-cache layout) | Claude Code handles API construction |
| `effective_prompt()` trailer mechanism | Agent .md files serve this role in Claude Code |
| Multi-arch container images | No Docker container |
| Fork PR support (`pull_request_target`) | No GitHub Actions workflow |
| Cache priming | No API-level control |
| Baked semgrep rulesets | No container; `--config=auto` appropriate for local |

## Files changed

| File | Change |
|---|---|
| `agents/adversarial-general.md` | Added missing knowledge-cutoff guardrail |
| `agents/architecture-reviewer.md` | Upgraded guardrail text |
| `agents/security-reviewer.md` | Upgraded guardrail text |
| `agents/blind-hunter.md` | Upgraded guardrail text |
| `agents/edge-case-hunter.md` | Upgraded guardrail text + path-tracing item |
| `scripts/run-trufflehog.sh` | Test-file demotion + single-invocation optimization |
| `suppressions.json` | Added trufflehog-unverified-test-file rule |
| `SEVERITY.md` | Updated trufflehog row, added checkov row |
| `SKILL.md` | Added ruby-org verify type, checkov Phase 1b integration |
| `README.md` | Added ruby-org to verify table, checkov to deterministic checks |
| `CLAUDE.md` | Added ruby-org to ecosystems list, checkov to file layout |
| `scripts/run-checkov.sh` | New file — checkov IaC scanner |
| `install.sh` | Added run-checkov.sh to installation list |

## Test plan

- [x] Verify adversarial-general guardrail block is present and correctly placed
- [x] Verify all 5 agent guardrails use identical standardized text (with edge-case-hunter's extra item)
- [x] Set `TRUFFLEHOG_MOCK_FILE` with unverified test-file findings → confirm Low/40 + tag
- [x] Verify verified secrets in test files remain Critical/95
- [x] Validate `run-checkov.sh` syntax (`bash -n`) and shellcheck
- [x] Validate `suppressions.json` is valid JSON
- [x] Verify ruby-org verify type documentation is consistent across SKILL.md, README.md, CLAUDE.md
- [x] `curl -sfI "https://cache.ruby-lang.org/pub/ruby/3.3/ruby-3.3.0.tar.gz"` returns 200
- [x] Run `./install.sh --local` and verify all files are installed
- [ ] Run `/comprehensive-review` on a branch with dependency bumps — no version-existence false positives

### CODE REVIEW

File: skills/comprehensive-review/scripts/run-checkov.sh [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      IAC_FILES+=("$file") ;;
+    Dockerfile|*/Dockerfile|Dockerfile.*|*/Dockerfile.*|*.dockerfile)
+      IAC_FILES+=("$file") ;;
+    *.yaml|*.yml)
+      # k8s sniff: apiVersion must match a k8s-shaped value — either bare
+      # "v<N>" (core API group) or "<group>/v<N>[alpha|beta<N>]" (named group).
+      # Plus kind must be present. Uses POSIX ERE (no backreferences).
+      if grep -qE '^[[:space:]]*AWSTemplateFormatVersion:' "$file" 2>/dev/null \
+         || grep -qE 'schema\.management\.azure\.com' "$file" 2>/dev/null \
+         || { grep -qE '^[[:space:]]*apiVersion:[[:space:]]*(([a-z0-9][-a-z0-9.]*/)?v[0-9]+(alpha[0-9]+|beta[0-9]+)?|"([a-z0-9][-a-z0-9.]*/)?v[0-9]+(alpha[0-9]+|beta[0-9]+)?")[[:space:]]*$' "$file" 2>/dev/null \
```
[the remark this answers is not in the document]
Comment (gchaix) [reply skills/comprehensive-review/scripts/run-checkov.sh:75]: Not changing — this is a verbatim port of ai-pr-review's regex (line 84 of `analyzers/run-checkov.sh`). The `$` anchor is a deliberate design decision documented in the comments above the regex: it prevents false positives on non-k8s YAML with free-form `apiVersion` values (OpenAPI schemas, custom tooling configs, CRD example docs). Trailing comments on `apiVersion` lines are vanishingly rare in k8s manifests. Loosening the anchor would cause checkov to spin up on more non-IaC YAML, which is the exact cold-start penalty this content sniff exists to avoid.

File: skills/comprehensive-review/scripts/run-checkov.sh [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    exit 0
+  fi
+  CHECKOV_OUTPUT=$(cat "$CHECKOV_MOCK_FILE")
+else
+  # Build --file arg list: checkov accepts repeated --file flags
+  CHECKOV_FILE_ARGS=()
+  for f in "${IAC_FILES[@]}"; do
+    CHECKOV_FILE_ARGS+=(--file "$f")
+  done
+
```
[the remark this answers is not in the document]
Comment (gchaix) [reply skills/comprehensive-review/scripts/run-checkov.sh:109]: Fixed in 0c1643d — removed `--compact` flag so `check_id_name` is present in the full JSON output. The fallback chain `check_id_name // .resource // "policy violation"` now works correctly since `check_id_name` is populated in non-compact mode.

File: skills/comprehensive-review/scripts/run-trufflehog.sh [hunk 3]
```diff
@@ -105,22 +134,24 @@ if [[ ${#TARGET_FILES[@]} -eq 0 ]]; then
...
-  TH_OUTPUT=$(trufflehog filesystem --json --no-update "$file" 2>/dev/null || true)
-  [[ -z "$TH_OUTPUT" ]] && continue
-
-  FILE_FINDINGS=$(echo "$TH_OUTPUT" | _th_transform 2>/dev/null) || {
-    echo "WARNING: trufflehog output for ${file} could not be parsed; skipping." >&2
-    continue
-  }
-  FILE_ARRAYS+=("$FILE_FINDINGS")
-done
+# Pass all target files to a single trufflehog invocation rather than forking
```
[the remark this answers is not in the document]
Comment (gchaix) [reply skills/comprehensive-review/scripts/run-trufflehog.sh:137]: False positive — `trufflehog filesystem` accepts variadic path arguments. This is the identical invocation pattern used in ai-pr-review production (line 121 of `analyzers/run-trufflehog.sh`), tested and shipped in v0.7.0. Added a clarifying comment in 0c1643d noting this is verified in production.
